pkg/sharings/sharings.go

```diff
@@ -0,0 +1,140 @@
+package sharings
+
+import (
+	"github.com/cozy/cozy-stack/pkg/consts"
+	"github.com/cozy/cozy-stack/pkg/couchdb"
+	"github.com/cozy/cozy-stack/pkg/permissions"
+	"github.com/cozy/cozy-stack/pkg/utils"
+	"github.com/cozy/cozy-stack/web/jsonapi"
+)
+
+// Sharing contains all the information about a sharing
+type Sharing struct {
+	SID         string `json:"_id,omitempty"`
+	SRev        string `json:"_rev,omitempty"`
+	Type        string `json:"type,omitempty"`
+	Owner       bool   `json:"owner"`
+	Desc        string `json:"desc,omitempty"`
+	SharingID   string `json:"sharing_id,omitempty"`
+	SharingType string `json:"sharing_type"`
+
+	Permissions *permissions.Set    `json:"permissions,omitempty"`
+	SRecipients []*SharingRecipient `json:"recipients,omitempty"`
```

Comment:
The code you were speaking about I guess is this https://github.com/cozy/cozy-stack/blob/749c557e354f134cb2b359ebf5736b88683d363b/pkg/vfs/file.go#L132-L142

In files, there is a "referenced_by" field, which is stored in couchdb, but we removed them from jsonapi (where they are relations instead).  

May be we mostly have a naming issue
- Recipient -> Contact
- SharingRecipient -> RecipientStatus

Not sure what is the proper JSON-API representation of relationships meta data (something with "meta" along side "links" within "relationships"."data" ???)